### PR TITLE
Fixes #1724. Configure module by Configuration object

### LINKED ISSUE #1724: Module configuration must be done by Configuration object.
At the moment module configuration (set configuration values to the module) is done by the Map<String, Objects> containing configuration id as key and configuration value corresponding to this key.
The configuration values has to be passed to the module as Configuration object.

### PR DESCRIPTION
**Pull request by danchom**
Signed-off-by: Yordan Mihaylov <j.mihailov@prosyst.com>

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        Configuration configuration = new Configuration();
+        JsonObject jo = (JsonObject) json;
+        for (Entry<String, JsonElement> e : jo.entrySet()) {
+            JsonPrimitive value = e.getValue().getAsJsonPrimitive();
+            if (value.isString()) {
+                configuration.put(e.getKey(), value.getAsString());
+            } else if (value.isNumber()) {
+                configuration.put(e.getKey(), value.getAsBigDecimal());
+            } else if (value.isBoolean()) {
+                configuration.put(e.getKey(), value.getAsBigDecimal());
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java]: Shouldn't this be value.getAsBoolean()?

File: bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.eclipse.smarthome.config.core.Configuration;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+
+public class ConfigurationDeserializer implements JsonDeserializer<Configuration> {
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationDeserializer.java]: please add javadoc

File: bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationSerializer.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package org.eclipse.smarthome.automation.parser.gson.internal;
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationSerializer.java]: please add license header

File: bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationSerializer.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.Configuration;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+public class ConfigurationSerializer implements JsonSerializer<Configuration> {
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.parser.gson/src/main/java/org/eclipse/smarthome/automation/parser/gson/internal/ConfigurationSerializer.java]: please add javadoc

File: protocols/enocean/org.eclipse.smarthome.protocols.enocean.basedriver.impl/.project [hunk 5]
```diff
@@ -6,6 +6,11 @@
 	</projects>
```
Comment (kaikreuzer) [inline protocols/enocean/org.eclipse.smarthome.protocols.enocean.basedriver.impl/.project]: please remove changes to this file

File: protocols/enocean/org.eclipse.smarthome.protocols.enocean.eeps.basic/.project [hunk 6]
```diff
@@ -6,6 +6,11 @@
 	</projects>
```
Comment (kaikreuzer) [inline protocols/enocean/org.eclipse.smarthome.protocols.enocean.eeps.basic/.project]: please remove changes to this file

File: protocols/enocean/org.eclipse.smarthome.protocols.enocean.sample.client/.project [hunk 7]
```diff
@@ -6,6 +6,11 @@
 	</projects>
```
Comment (kaikreuzer) [inline protocols/enocean/org.eclipse.smarthome.protocols.enocean.sample.client/.project]: please remove changes to this file

**Comment by kaikreuzer** [comment]
thanks, lgtm besides the small comments.

**Comment by danchom** [comment]
Fix comments

**Comment by kaikreuzer** [comment]
Thanks!

**Pull request merged**